In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt  

In [13]:
results_cpu = pd.read_csv('./benchmarks/random_forest/results11_cpu.csv')[1:]
results_tpu = pd.read_csv('./benchmarks/random_forest/results11_edgetpu.csv')[1:]

results_less_tpu = pd.read_csv('./benchmarks/random_forest/results11_less_tpu.csv')[1:]

In [14]:
mean_cpu = results_cpu['TimeMS'].mean()
mean_tpu = results_tpu['TimeMS'].mean()

mean_less_tpu = results_less_tpu['TimeMS'].mean()

In [16]:
print(f'Mean CPU:\t\t{mean_cpu : .2f}ms\nMean TPU:\t\t{mean_tpu : .2f}ms')
print(f'~ Time for tpu opps:\t{mean_less_tpu : .2f}ms')

Mean CPU:		 0.03ms
Mean TPU:		 1.04ms
~ Time for tpu opps:	 0.53ms


### Ergebnis
Die volle Inferenz auf der CPU dauert $0.03$ ms. <br>
Nimmt man die TPU dazu benötigt man $1.04$ ms. <br>
Das Entfernen der $8$ CPU Operationen ermöglicht das volle Mappen auf die Edge TPU. <br> 
Die $8$ Operationen auf der Edge TPU benötigen $0.53$ ms. <br> 
Also benötigt die Kommunikation zwischen TPU und CPU in etwa $0.51$ ms.